roles/cds/tasks/cds_instance.yml

```diff
@@ -0,0 +1,108 @@
+# WLP_OUTPUT_DIR = the place it writes random temp files and whatnot
+
+- name: Set facts from the include context
+  ansible.builtin.set_fact:
+    cds_instance_name: "{{ cds_name }}"
+    cds_http_port: "{{ cds_instance.http_port }}"
+    cds_https_port: "{{ cds_instance.https_port }}"
+    cds_contests: "{{ cds_instance.contests }}"
+    cds_accounts: "{{ cds_instance.accounts }}"
+    cds_needs_restart: "{{ needs_restart }}"
+
+- name: Create CDS directories
+  ansible.builtin.file:
+    path: /var/lib/cds/{{ cds_instance_name }}
+    state: directory
+    owner: cds
+    group: cds
+    mode: '0755'
+
+- name: Make sure directories exist
+  ansible.builtin.file:
+    state: directory
+    owner: cds
+    group: cds
+    mode: '0755'
+    path: /srv/cds/wlp/usr/servers/{{ cds_instance_name }}/{{ item }}
+  with_items:
+    - apps
+    - config
+
+- name: Symlink CDS war to the instance
+  ansible.builtin.file:
+    state: link
+    force: true
+    src: /srv/cds/CDS.war
+    dest: /srv/cds/wlp/usr/servers/{{ cds_instance_name }}/apps/CDS.war
+
+- name: Populate CDS server.xml
+  ansible.builtin.template:
+    src: server.xml.j2
+    dest: /srv/cds/wlp/usr/servers/{{ cds_instance_name }}/server.xml
+    owner: cds
+    group: cds
+    mode: '0644'
+  register: server_updated
+
+- name: Update cds_needs_restart (after server.xml)
+  ansible.builtin.set_fact:
+    cds_needs_restart: "{{ cds_needs_restart | bool or server_updated.changed }}"
+
+- name: Populate CDS accounts.yaml
+  ansible.builtin.template:
+    src: accounts.yaml.j2
+    dest: /srv/cds/wlp/usr/servers/{{ cds_instance_name }}/config/accounts.yaml
+    owner: cds
+    group: cds
+    mode: '0600'
+  register: accounts_updated
+
+- name: Update cds_needs_restart (after accounts.yaml)
+  ansible.builtin.set_fact:
+    cds_needs_restart: "{{ cds_needs_restart | bool or accounts_updated.changed }}"
+
+- name: Populate CDS cdsConfig.xml
+  ansible.builtin.template:
+    src: cdsConfig.xml.j2
+    dest: /srv/cds/wlp/usr/servers/{{ cds_instance_name }}/config/cdsConfig.xml
+    owner: cds
+    group: cds
+    mode: '0600'
+  register: cds_config_updated
+
+- name: Update cds_needs_restart (after cdsConfig.xml)
+  ansible.builtin.set_fact:
+    cds_needs_restart: "{{ cds_needs_restart | bool or cds_config_updated.changed }}"
+
+- name: Create contest specific directory
+  ansible.builtin.file:
+    path: /srv/cds/contests/{{ item.path }}
+    state: directory
+    owner: cds
+    group: cds
+    mode: '0755'
+  loop: "{{ cds_contests }}"
+
+- name: Copy cds systemd unit file
+  ansible.builtin.template:
+    src: templates/cds.service.j2
+    dest: /etc/systemd/system/{{ cds_instance_name }}.service
+    mode: '0644'
+  register: unit_updated
+
+- name: "Update cds_needs_restart: after service for {{ cds_instance_name }}"
+  ansible.builtin.set_fact:
+    cds_needs_restart: "{{ cds_needs_restart | bool or unit_updated.changed }}"
+
+- name: Restart+enable cds instance
```

Comment:
Why can't this be done in the handler anymore? The templates will do a notify on change, but what is the resource where the notify route does not work?

Reply:
There's no way to parameterize a handler, so notify doesn't work. We include this tasks file multiple times, once per instance (e.g. a normal cds instance and a p100 instance), but you can't make dynamic handlers (at least that I could find online). And you don't want to restart the regular CDS if you change the config for p100 so you can't just restart all of the instances.